docs/options.md

```diff
@@ -176,6 +176,32 @@ const md = require('markdown-it')({ html: true })
 }
 ```
 
+If you want to specify it further, you can also pass a configuration object to the `markdown` option, which will be passed directly to `markdown-it`. See [their documentation](https://github.com/markdown-it/markdown-it#init-with-presets-and-options) for available options.
```

Comment:
As the interface, I don't prefer to expect markdown-it for "object" implicitly. How's about changing to a more explicit way like having `markdownIt` option for the top level?

I'd to keep the possibility to replace the default markdown compiler without any breaking changes.

Reply:
I agree with your general idea, but having `markdownIt` and `markdown` options might confuse some folk. If we go with this, we should at least be very careful in the way we document how it's used.

As I was implementing this yesterday, I started playing with the many plugins available to `markdown-it` as well. These can be passed like so:

```js
// first, import
const markdownIt = require('markdown-it');

// second, set up with the options
const compiler = markdownIt(options.markdown);

// third, hook up any plugins.
compiler.use(prism);
compiler.use(somethingElse);
```

Should we support this as well, or should we ask people to use the `markdown` option for this kind of usecase? 

Here's an example of the API we _could_ go for: We could add an extra option to `markdownIt` called `use`, and this could accept an array of plugins, like so:

```
{
  options: {
    markdownIt: {
      breaks: true,
      use: [prism, anotherPlugin]
    }
  }
}
```

I'd be willing to do the work no matter - I'll let you as the project owner decide the direction :)

Reply:
> but having markdownIt and markdown options might confuse some folk

Totally agree. That's my concern too. I prefer to kill `markdown` option if `markdownIt` option is provided. And I'm thinking about migrating to TypeScript to serve typedefs for the loader's options eventually.

> `use`

Your idea expects the user's additional knowledge since it's adding new syntax.
So how's about this?:

- `markdownIt` option supports 2 ways of configuration
    - The result of `require('markdown-it')(/*whatever*/).use(...).use(...)` -> loader just calls `render`
    - The options for `require('markdown-it')` -> loader instantiate `markdown-it`

This enables just copy+paste of configuration for `markdown-it` from other places 😉  I'm not sure how we detect the differences between those 2 cases technically yet though.

Reply:
Hm... How would an example use of option 1 look?

Reply:
```
{
  options: {
    markdownIt: require('markdown-it').(myConfig).use(prism).use(anotherPlugin)
  }
}
```

```
{
  options: {
    markdownIt: { ...myConfig }
  }
}
```

so, if you need to use `use`, should pick the former.

Reply:
Ah okay. Yeah this works fine!